common/src/main/java/com/microsoft/identity/common/internal/cache/SharedPreferencesFileManagerSimpleCacheImpl.java

```diff
@@ -0,0 +1,190 @@
+//  OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+//  THE SOFTWARE.
+package com.microsoft.identity.common.internal.cache;
+
+import android.content.Context;
+import android.os.Build;
+import android.os.SystemClock;
+import android.util.Log;
+
+import androidx.annotation.NonNull;
+
+import com.google.gson.Gson;
+import com.microsoft.identity.common.internal.util.StringUtil;
+import com.microsoft.identity.common.logging.Logger;
+
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+import java.util.concurrent.Callable;
+import java.util.concurrent.TimeUnit;
+
+import lombok.SneakyThrows;
+
+/**
+ * A simple metadata store definition that uses SharedPreferencesFileManager to persist, read,
+ * update, and delete data. Please note that all CRUD actions return success, as the underlying
+ * store's API does not surface a success indicator. If you need stronger guarantees that an
+ * operation was successful, use {@link SharedPreferencesSimpleCacheImpl} which is less performance
+ * oriented.
+ * <p>
+ * Data serializes as JSON.
+ *
+ * @param <T> The type of metadata that will be persisted.
+ * @see SharedPreferencesSimpleCacheImpl
+ */
+public abstract class SharedPreferencesFileManagerSimpleCacheImpl<T> implements ISimpleCache<T>, IListTypeToken {
+
+    private static final String TAG = SharedPreferencesFileManagerSimpleCacheImpl.class.getSimpleName();
+    private static final String EMTPY_ARRAY = "[]";
+    private static final String TIMING_TAG = "execWithTiming";
+
+    private ISharedPreferencesFileManager mSharedPrefsFileManager;
+    private final String mKeySingleEntry;
+    private final Gson mGson = new Gson();
+
+    public SharedPreferencesFileManagerSimpleCacheImpl(@NonNull final Context context,
+                                                       @NonNull final String prefsName,
+                                                       @NonNull final String singleKey) {
+        Logger.verbose(
+                TAG + "::ctor",
+                "Init"
+        );
+        mSharedPrefsFileManager = SharedPreferencesFileManager.getSharedPreferences(
+                context,
+                prefsName,
+                Context.MODE_PRIVATE,
+                null // File is not encrypted
+        );
+        mKeySingleEntry = singleKey;
+    }
+
+    private interface NamedRunnable<V> extends Callable<V> {
+        String getName();
+    }
+
+    @SneakyThrows
+    private <V> V execWithTiming(@NonNull final NamedRunnable<V> runnable) {
+        final TimeUnit timeUnit;
+        final long startTime;
+        final long execTime;
+
+        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.JELLY_BEAN_MR1) {
+            startTime = SystemClock.elapsedRealtimeNanos();
+            timeUnit = TimeUnit.NANOSECONDS;
+        } else {
+            startTime = SystemClock.elapsedRealtime();
+            timeUnit = TimeUnit.MILLISECONDS;
+        }
+
+        final V v = runnable.call();
```

Comment:
could/should consider:

try {
 return runnable.call();
} finally {
 //logging
}

This way we get the logs even if there's an exception.